### DISCUSSION TITLE
Modbus TCP slave multiple clients?

### DISCUSSION THREAD

**Discussion by nated0g** [alisitsyn/modbus_support#1] [Q&A] [answered]
Hello @alisitsyn!

I'm not sure if this was actually the idea behind this repo, but here I  am anyway. 

Thanks for the hard work getting Modbus TCP into ESP-IDF.  I am finally getting around to rewriting the codebase for a couple projects and had a question.

Will the Modbus TCP slave library in ESP-IDF 4.2 support connections from multiple masters to one slave device?  I wrote an implementation combining Arduino Modbus and AsyncTCP libraries, but I am rewriting my entire codebase using only ESP-IDF and wanted to know if this was supported out of the box.

Thanks again,
nated0g

**Comment by alisitsyn** [comment, marked as answer]
Hello @nated0g,

Thank you for your question and interest to ESP-MODBUS. The [Modbus TCP slave library](https://github.com/espressif/esp-idf/tree/master/components/freemodbus/tcp_slave) supports connection from multiple masters and it is supported out of the box. The `CONFIG_FMB_TCP_PORT_MAX_CONN` kconfig option allows configuring maximum number of connection from masters to the same slave device. It is not recommended establishing more than 5 connections to provide reliable communication because it uses enough CPU resources, memory, throughput.

Let me know if you have questions.

**Reply by nated0g** [reply]
Hi @alisitsyn,

This is excellent.  Thank you again for your working getting this into the IDF.
